packages/cli/src/start.ts

```diff
@@ -0,0 +1,217 @@
+import type { Config as CLIConfig } from "@react-native-community/cli-types";
+import {
+  createDevServerMiddleware,
+  indexPageMiddleware,
+} from "@react-native-community/cli-server-api";
+import {
+  createTerminal,
+  loadMetroConfig,
+  startServer,
+} from "@rnx-kit/metro-service";
+import { Service } from "@rnx-kit/typescript-service";
+import chalk from "chalk";
+import type { Reporter, ReportableEvent } from "metro";
+import type Server from "metro/src/Server";
+import type { Middleware } from "metro-config";
+import path from "path";
+import readline from "readline";
+import { getKitServerConfig } from "./kit-config";
+import { customizeMetroConfig } from "./metro-config";
+import type { TSProjectInfo } from "./types";
+
+export type CLIStartOptions = {
+  host: string;
+  port: number;
+  projectRoot?: string;
+  watchFolders?: string[];
+  assetPlugins?: string[];
+  sourceExts?: string[];
+  maxWorkers?: number;
+  customLogReporterPath?: string;
+  https?: boolean;
+  key?: string;
+  cert?: string;
+  resetCache?: boolean;
+  config?: string;
+  interactive: boolean;
+};
+
+export async function rnxStart(
+  _argv: Array<string>,
+  cliConfig: CLIConfig,
+  cliOptions: CLIStartOptions
+): Promise<void> {
+  const serverConfig = getKitServerConfig(cliOptions);
+  if (!serverConfig) {
+    return Promise.resolve();
+  }
+
+  // interactive mode requires raw access to stdin
+  let interactive = cliOptions.interactive;
+  if (interactive) {
+    interactive = Boolean(process.stdin.setRawMode);
```

Comment:
If we're checking whether a text terminal is attached, we should probably use `isTTY` instead:
```suggestion
    interactive = Boolean(process.stdin.isTTY);
```